.github/workflows/haskell.yml

```diff
@@ -0,0 +1,161 @@
+name: Haskell CI
+
+on:
+  push:
+  create:
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        # Add more elements to this list to run multiple instances of the build in CI.  Increasing the
+        # number instances is a good way to trigger flaky build failures
+        n: [1]
+
+        ghc: ["8.10.7"]
+        os: [ubuntu-latest, macos-latest, windows-latest]
+
+    env:
+      # OpenSSL is installed in a non-standard location in MacOS. See
+      # https://github.com/actions/virtual-environments/blob/main/images/macos/macos-latest-Readme.md
+      PKG_CONFIG_PATH: ${{ (matrix.os == 'macos-latest' && '/usr/lib/pkgconfig:/usr/local/opt/openssl@1.1/lib/pkgconfig') || (matrix.os == 'ubuntu-latest' && '/usr/lib/pkgconfig:/usr/local/lib/pkgconfig') || '' }}
+      
+      # FIXME: this is arguably a bug, and pkg-config should return the right values!
+      LD_LIBRARY_PATH: ${{ (matrix.os != 'windows-latest' && '/usr/local/lib') || '' }}
+
+    steps:
+    - name: "WIN: Install System Dependencies via pacman (msys2)"
+      if: runner.os == 'Windows'
+      run: |
+         # ghcup should be installed on current GHA Windows runners. Let's use ghcup to run
+         # pacman, to install the necessary dependencies, ...
+         ghcup run -- pacman --noconfirm -S `
+            mingw-w64-x86_64-pkg-config `
+            mingw-w64-x86_64-openssl `
+            base-devel `
+            autoconf-wrapper `
+            autoconf `
+            automake `
+            libtool `
+            make
+  
+    # this seems to break something. It _must_ come after the pacman setup
+    # above. It appears as if PATHEXT is set _after_ ghcup install ghc/cabal, and
+    # as such we'd need pacman.exe instead.
+    - name: Setup Haskell
+      run: |
+        # Use GHCUP to manage ghc/cabal
+        ghcup install ghc --set ${{ matrix.ghc }}
+        ghcup install cabal --set 3.6.2.0
+
+        ghc --version
+        cabal --version
+
+    - name: "WIN: fixup cabal config"
```

Comment:
There's seems to be a more simple way to setup GHC on windows/mac using GitHub actions
https://github.com/haskell/text-icu/blob/master/.github/workflows/cabal-mac-win.yml
Maybe it could be used here?